sv/Vehicle.py

```diff
@@ -347,6 +350,39 @@ def tick(self, tick_count, delta_time, sim_time):
             traffic_vehicles_orp = {},
         )
         vehicle_path = [(n.x, n.y) for n in self.path]
+
+        if self.collision_vid is not None:
+            # Try to get the collision vehicle, if not found, raise a KeyError
+            collision_vehicle = None
+            try:
+                collision_vehicle = self.scenario_vehicles[self.collision_vid]
+                vehicle_pos = np.array([collision_vehicle.state.x, collision_vehicle.state.y])
+            except KeyError:
+                raise KeyError(f"collision vehicle with vid {self.collision_vid} not found in scenario")
+
+            # Calculate time to collision if collision vehicle is found
+            if collision_vehicle:
+                if self.collision_point is not None:
+                    collision_pt = [self.collision_point.x, self.collision_point.y]
+                    collision_segment_prev_node, collision_segment_next_node = self.get_curr_and_prev_path_nodes(self.path)
+
+                    # Euclidean distance between vehicle and collision point
+                    collision_vehicle_dist_to_collision = np.sqrt(np.sum((collision_pt - vehicle_pos) ** 2))
+                    time_to_collision = collision_vehicle_dist_to_collision / collision_vehicle.state.s_vel
+                else:
```

Comment:
well, if there's no explicit designated collision point, it should be discovered as before when an agent can adjust it's speed. Otherwise, it's breaking pedestrians.